packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-has-one-relationship/BookUpdateForm.jsx

```diff
@@ -0,0 +1,391 @@
+                    event.stopPropagation();
+                    removeItem(index);
+                  }}
+                />
+              </Badge>
+            );
+          })}
+        </ScrollView>
+      )}
+      <Divider orientation="horizontal" marginTop={5} />
+    </React.Fragment>
+  );
+
+  // should not render input if limit reached & not editing
+  if (lengthLimit !== undefined && items.length >= lengthLimit && !isEditing) {
+    return arraySection;
+  }
+
+  return (
+    <React.Fragment>
+      {isEditing && children}
+      {!isEditing ? (
+        <>
+          <Text>{label}</Text>
+          <Button
+            onClick={() => {
+              setIsEditing(true);
+            }}
+          >
+            Add item
+          </Button>
+        </>
+      ) : (
+        <Flex justifyContent="flex-end">
+          {(currentFieldValue || isEditing) && (
+            <Button
+              children="Cancel"
+              type="button"
+              size="small"
+              onClick={() => {
+                setFieldValue(defaultFieldValue);
+                setIsEditing(false);
+                setSelectedBadgeIndex(undefined);
+              }}
+            ></Button>
+          )}
+          <Button
+            size="small"
+            variation="link"
+            color={tokens.colors.brand.primary[80]}
+            isDisabled={hasError}
+            onClick={addItem}
+          >
+            {selectedBadgeIndex !== undefined ? 'Save' : 'Add'}
+          </Button>
+        </Flex>
+      )}
+      {arraySection}
+    </React.Fragment>
+  );
+}
+export default function BookUpdateForm(props) {
+  const { id, book, onSuccess, onError, onSubmit, onCancel, onValidate, onChange, overrides, ...rest } = props;
+  const initialValues = {
+    name: undefined,
+    primaryAuthor: undefined,
+  };
+  const [name, setName] = React.useState(initialValues.name);
+  const [primaryAuthor, setPrimaryAuthor] = React.useState(initialValues.primaryAuthor);
+
+  // grab records for suggestion
+  const authorRecords = useDataStoreBinding({
+    type: 'collection',
+    model: Author,
+  }).items;
+
+  // generated from valueMappings config
+  const getDisplayValue = {
+    primaryAuthor: (record) => record?.name,
+  };
+
+  const [errors, setErrors] = React.useState({});
+  const resetStateValues = () => {
+    setName(initialValues.name);
+    setPrimaryAuthor(primaryAuthor);
+    setCurrentPrimaryAuthorValue(undefined);
+    setCurrentPrimaryAuthorDisplayValue(undefined);
+  };
+
+  // fetch model data to fill form with preset values
+  const [bookRecord, setBookRecord] = React.useState(book);
+  React.useEffect(() => {
+    const queryData = async () => {
+      const record = id ? await DataStore.query(Book, id) : book;
+      setBookRecord(record);
+      // get existing relationship record
+      const existingAuthor = record ? await DataStore.query(Author, (a) => a.id.eq(record.authorId)) : primaryAuthor;
```

Comment:
Should this fall back to `book.author`? Wouldn't `primaryAuthor` always be undefined here?

Reply:
If it can't find an existing relationship record, it should fallback to undefined.  I'll update it  from `primaryAuthor` to  `undefined` to be clearer